logging/interceptor.go

```diff
@@ -39,6 +62,161 @@ func LogLevelInterceptor(defaultLevel logrus.Level) grpc.UnaryServerInterceptor
 	}
 }
 
+func UnaryClientInterceptor(logger *logrus.Logger, opts ...Option) grpc.UnaryClientInterceptor {
+	options := initOptions(opts)
+
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		startTime := time.Now()
+		fields := newLoggerFields(method, startTime, DefaultClientKindValue)
+
+		ctx = setInterceptorFields(ctx, fields, logger, options, startTime)
+
+		err := invoker(ctx, method, req, reply, cc, opts...)
+		if err != nil {
+			fields[logrus.ErrorKey] = err
+		}
+
+		code := status.Code(err)
+		fields[DefaultGRPCCodeKey] = code.String()
+
+		levelLogf(
+			logrus.NewEntry(logger).WithFields(fields),
+			options.codeToLevel(code),
+			"finished unary call with code "+code.String())
+
+		return err
+	}
+}
+
+func UnaryServerInterceptor(logger *logrus.Logger, opts ...Option) grpc.UnaryServerInterceptor {
```

Comment:
If this were implemented as a [`ChainUnaryServer`](https://godoc.org/github.com/grpc-ecosystem/go-grpc-middleware#ChainUnaryServer), it could be composed of existing interceptors, like the requestid middleware, without having to reimplement them.